src/entry.h

```diff
@@ -73,6 +73,7 @@ void entryFree(entry *entry);
 
 /* Creates a new entry with the given field, value, and optional expiry. */
 entry *entryCreate(const_sds field, sds value, long long expiry);
+entry *entryUpdateAsStringRef(entry *entry, const char *buf, size_t len, long long expiry);
```

Comment:
This feature is fundamentally changing the functionality and structure of the "entry".  The header comments need to be updated in both the .h and .c files.